### ISSUE TITLE
Registering through /register gives two Riot bots if two tabs kept open

### ISSUE THREAD

**Issue by lukebarnard1** [vector-im/riot-web#4351]
We should only invite Riot bot once

**Comment by richvdh** [comment]
Also: the unfocussed tab ends up with a stuck spinner - I think because the `createRoom` at https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/createRoom.js#L85 finishes before the spinner gets created at https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/createRoom.js#L81.

**Comment by lukebarnard1** [comment]
@lukebarnard1 wonders why on _earth_ there's a `setTimeout(..., 0)` there :(

**Comment by lukebarnard1** [comment]
>We should only invite Riot bot once

Doing this will _slightly_ non-trivial because we don't have idempotent inviting of users, so we can't just spam /invite `@Riot-Bot:matrix.org` and expect a single 1:1 with Riot Bot.

The only instance that must be opened for the process to complete is the one with email tokens in it. So we can inspect the query params before inviting Riot Bot. This only works when we're registering with an email address. It's possible that this flow is not needed for a given home server. So we can't use just use the presence of the email tokens to determine whether we should invite the Riot Bot. 

Basically anything client-side will have the same problem..

**Comment by turt2live** [comment]
We've since fixed an eerily similar instance of this in our auth rework. If it continues to happen, please open a new issue.
